legal-api/src/legal_api/services/filings/validations/registration.py

```diff
@@ -0,0 +1,131 @@
+# Copyright © 2022 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validation for the Registration filing."""
+from datetime import timedelta
+from http import HTTPStatus  # pylint: disable=wrong-import-order
+from typing import Dict, Final, Optional
+
+import pycountry
+from dateutil.relativedelta import relativedelta
+from flask_babel import _ as babel  # noqa: N813, I004, I001, I003
+from legal_api.errors import Error
+from legal_api.models import Business, PartyRole
+from legal_api.utils.legislation_datetime import LegislationDatetime
+
+from ...utils import get_date, get_str
+from .common_validations import validate_court_order
+
+
+def validate(registration_json: Dict) -> Optional[Error]:
+    """Validate the Registration filing."""
+    if not registration_json:
+        return Error(HTTPStatus.BAD_REQUEST, [{'error': babel('A valid filing is required.')}])
+    legal_type = get_str(registration_json, '/filing/business/legalType')
```

Comment:
I just noticed in the registration schema that we don't validate the business block like we do for IA.  I think we need to add this?  Even there, the business block only requires the `identifier`.  We may need to throw a validation error here if no legal type is provided I think.

https://github.com/bcgov/business-schemas/blob/394735852686aab57cf405def29433b5b5dbea01/src/registry_schemas/schemas/filing.json#L211-L228